src/libwolfboot.c

```diff
@@ -787,10 +882,16 @@ static int RAMFUNCTION hal_set_key(const uint8_t *k, const uint8_t *nonce)
     uint32_t addr_align = addr & (~(WOLFBOOT_SECTOR_SIZE - 1));
     uint32_t addr_off = addr & (WOLFBOOT_SECTOR_SIZE - 1);
     int ret = 0;
+    int sel_sec = 0;
+#ifdef NVM_FLASH_WRITEONCE
+    sel_sec = nvm_select_fresh_sector(PART_BOOT);
+    addr_align -= (sel_sec * WOLFBOOT_SECTOR_SIZE);
```

Comment:
Should we check that `addr_align > (sel_sec * WOLFBOOT_SECTOR_SIZE)`?

Reply:
not sure. This comes from the constants at the beginning of the function, marking the end of the boot partition. Also, on most architectures the address of the boot partition does not start from 0, so the check will be true anyway even if the BOOT partition is that small?